app/controllers/auth_source_ldaps_controller.rb

```diff
@@ -18,6 +20,31 @@ def create
     end
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(10) do
+        temp_auth_source_ldap = AuthSourceLdap.new(params[:auth_source_ldap])
+        temp_auth_source_ldap.ldap_con.valid_user?("")
+      end
+      result[:success] = true
+      result[:message] = _("Test connection to LDAP server was successful.")
+    rescue Timeout::Error => exception
```

Comment:
I think you can do just `Timeout::Error, Net::LDAP::Error => exception` here, no need to handle them in separate contexts if you're going to do the same for all of them.